include/seqan3/core/debug_stream/range.hpp

```diff
@@ -93,12 +93,27 @@ namespace seqan3
  * to avoid ambiguous function calls.
  * \endif
  */
+#if 0
 template <typename char_t, std::ranges::input_range rng_t>
 inline debug_stream_type<char_t> & operator<<(debug_stream_type<char_t> & s, rng_t && r)
     requires detail::debug_stream_range_guard<rng_t>
 {
+#else
+
+// e.g. std::filesystem
+template <typename rng_t>
+concept nonrecursive_range = !std::same_as<std::remove_cvref_t<std::ranges::range_reference_t<rng_t>>, std::remove_cvref_t<rng_t>>;
+
+template <typename rng_t>
+    requires std::ranges::input_range<rng_t> && nonrecursive_range<rng_t>
```

Comment:
This shows that we don't need any "guard" requirement helper, since we can define printer logic locally and don't need to think about the possible global "state" (i.e. other types that might fulfil the same concept but should not be taken).